### ISSUE TITLE
cannot add element to high order map

### ISSUE THREAD

**Issue by SleepyRoy** [vlang/v#8171]
**V version:** 
master

**OS:** 
win10

**What did you do?**
```V
fn main() {
	mut m := map[string]map[string]int
	m['hello']['hi'] = 1
	println(m)
}
```

**What did you expect to see?**
`{'hello': {'hi': 1}}`
 
**What did you see instead?**
`{ }`

**Comment by kruzus** [comment]
Hello @SleepyRoy! I tried your snippet and got the same result. I tried running in different ways but no luck until I messed around with it, which it worked?
```v
fn main() {
	mut m := map[string]map[string]int
	mut a:= map[string]int
	a["second"] = 3
	m["first"] = a
	println(m)
}
```

Output:
`{'first': {'second': 3}}`


Obviously this isn't intended but I am kind of curious why, we cant just do `mut m := map[string]map[string]int`

```v
fn main() {
	mut m := map[string]map[string]int
	m['hello']['hi'] = 1
	println(m)
}
```
is not giving us: `{'hello': {'hi': 1}}`

cheers!
